integrated_channels/sap_success_factors/utils.py

```diff
@@ -5,32 +5,164 @@
 UNIX_EPOCH = datetime.datetime(1970, 1, 1, tzinfo=timezone.utc)
 UNIX_MIN_DATE_STRING = '1970-01-01T00:00:00Z'
 UNIX_MAX_DATE_STRING = '2038-01-19T03:14:07Z'
+SUCCESSFACTORS_OCN_LANGUAGE_CODES = {
+    "ms": {
+        "_": "Malay"
+    },
+    "ar": {
+        "_": "Arabic"
+    },
+    "bg": {
+        "_": "Bulgarian"
+    },
+    "cs": {
+        "_": "Czech"
+    },
+    "cy": {
+        "_": "Welsh"
+    },
+    "da": {
+        "_": "Danish"
+    },
+    "de": {
+        "_": "German"
+    },
+    "el": {
+        "_": "Greek"
+    },
+    "en": {
+        "ca": "English Canadian",
+        "gb": "English United Kingdom",
+        "_": "English"
+    },
+    "es": {
+        "mx": "Spanish Mexican",
+        "_": "Spanish"
+    },
+    "fi": {
+        "_": "Finnish"
+    },
+    "fr": {
+        "ca": "French Canadian",
+        "_": "French"
+    },
+    "hi": {
+        "_": "Hindi"
+    },
+    "hr": {
+        "_": "Croatian"
+    },
+    "hu": {
+        "_": "Hungarian"
+    },
+    "in": {
+        "_": "Indonesian"
+    },
+    "it": {
+        "_": "Italian"
+    },
+    "iw": {
+        "_": "Hebrew"
+    },
+    "ja": {
+        "_": "Japanese"
+    },
+    "ko": {
+        "_": "Korean"
+    },
+    "nl": {
+        "_": "Dutch"
+    },
+    "no": {
+        "_": "Norwegian"
+    },
+    "pl": {
+        "_": "Polish"
+    },
+    "pt": {
+        "br": "Brazilian Portuguese",
+        "_": "Portuguese"
+    },
+    "ro": {
+        "_": "Romanian"
+    },
+    "ru": {
+        "_": "Russian"
+    },
+    "sk": {
+        "_": "Slovak"
+    },
+    "sl": {
+        "_": "Slovenian"
+    },
+    "sr": {
+        "_": "Serbian"
+    },
+    "sv": {
+        "_": "Swedish"
+    },
+    "th": {
+        "_": "Thai"
+    },
+    "tr": {
+        "_": "Turkish"
+    },
+    "uk": {
+        "_": "Ukrainian"
+    },
+    "vi": {
+        "_": "Vietnamese"
+    },
+    "zh": {
+        "hk": "Chinese Hong Kong",
+        "tw": "Chinese Taiwan",
+        "_": "Chinese"
+    }
+}
 
 
-class SapCourseExporter(BaseCourseExporter):
+class SapCourseExporter(BaseCourseExporter):  # pylint: disable=abstract-method
     """
     Class to provide data transforms for SAP SuccessFactors course export task.
     """
 
     def __init__(self, user, plugin_configuration):
         super(SapCourseExporter, self).__init__(user, plugin_configuration)
 
-    def get_serialized_data(self):
-        final_structure = {
-            'ocnCourses': self.courses
-        }
-        return json.dumps(final_structure, sort_keys=True).encode('utf-8')
+    def get_serialized_data_blocks(self):
+        """
+        Return serialized blocks of data representing the courses to be POSTed, 1000 at a time.
+
+        Yields:
+            bytes: JSON-serialized course metadata structure
+            int: Number of records in this batch
+        """
+        page_length = 1000
+        location = 0
+        while (location < len(self.courses)) or not location:
```

Comment:
Performance nit -- you can set len(self.courses) as a variable prior to the loop and then you will not have to re-evaluate the length for every iteration -- this will be helpful especially if we are talking about 1000s of courses.